src/modules/fancyzones/FancyZonesLib/FancyZones.cpp

```diff
@@ -266,27 +265,25 @@ FancyZones::VirtualDesktopChanged() noexcept
     PostMessage(m_window, WM_PRIV_VD_SWITCH, 0, 0);
 }
 
-std::pair<winrt::com_ptr<IWorkArea>, std::vector<size_t>> FancyZones::GetAppZoneHistoryInfo(
-    HWND window,
-    HMONITOR monitor,
-    std::unordered_map<HMONITOR, winrt::com_ptr<IWorkArea>>& workAreaMap) noexcept
-{
-    if (workAreaMap.contains(monitor))
-    {
-        auto workArea = workAreaMap[monitor];
-        workAreaMap.erase(monitor); // monitor processed, remove entry from the map
-        return { workArea, workArea->GetWindowZoneIndexes(window) };
-    }
-    return { nullptr, {} };
-}
-
 std::pair<winrt::com_ptr<IWorkArea>, std::vector<size_t>> FancyZones::GetAppZoneHistoryInfo(HWND window, HMONITOR monitor, bool isPrimaryMonitor) noexcept
 {
     std::pair<winrt::com_ptr<IWorkArea>, std::vector<size_t>> appZoneHistoryInfo{ nullptr, {} };
     auto workAreaMap = m_workAreaHandler.GetWorkAreasByDesktopId(m_currentDesktopId);
+    if (workAreaMap.empty())
+    {
+        Logger::debug(L"No work area for the current desktop.");
+    }
 
     // Search application history on currently active monitor.
-    appZoneHistoryInfo = GetAppZoneHistoryInfo(window, monitor, workAreaMap);
+    if (workAreaMap.contains(monitor))
+    {
+        auto workArea = workAreaMap[monitor];
+        appZoneHistoryInfo = { workArea, workArea->GetWindowZoneIndexes(window) };
+    }
+    else
+    {
+        Logger::debug(L"No work area for the current desktop.");
```

Comment:
It would be good that this log message and the one at line 274 differ at least a bit. Eventual debugging in the future would benefit out of it :)

Reply:
You're right :)